### DISCUSSION TITLE
No identity secret

### DISCUSSION THREAD

**Discussion by LukaszSzalek-TomTom** [tomkerkhove/promitor#1879] [Q&A] [answered]
hi
i am trying to use promitor discovery and i am getting
Azure Landscape           │ Success │ Everything is well-configured.         ┃
┃ Resource Discovery Groups │ Success │ Everything is well-configured.         ┃
┃ Azure Authentication      │ Failed  │ Validation failed:                     ┃
┃                           │         │ Azure authentication is not configured ┃
┃                           │         │ correctly - No identity secret was     ┃
┃                           │         │ configured for service principle       ┃
┃                           │         │ authentication  

Setup:
![image](https://user-images.githubusercontent.com/79637749/145572793-23cad178-0682-4df9-9708-b86909ce848c.png)

I tried to use azureAuthentication:
  appId: ****
  appKey: ***
and 
azureAuthentication:
  appId: "" # [Deprecated] Prefer identity.id
  appKey: "" # [Deprecated] Prefer identity.key
  mode: "ServicePrincipal"
  identity:
    id: ""
    key: ""
    binding: ""

but no luck

Can you tell me how to pass secret key to app?

**Comment by tomkerkhove** [comment]
How did you install it? Manually or through Helm?

Because the authentication key should be in the `PROMITOR_AUTH_APPID` environment variable, not the YAML file

**Comment by LukaszSzalek-TomTom** [comment]
Via helm
so why in config i can set a key?
some missmatch between documentation and code?
how to set PROMITOR_AUTH_APPID  when app is running on pod already?

**Reply by tomkerkhove** [reply]
Can you share your Helm install command and config please?

**Comment by LukaszSzalek-TomTom** [comment]
as stated on  page
https://docs.promitor.io/deployment/resource-discovery/
helm repo add promitor https://charts.promitor.io/
then
❯ helm install promitor-agent-resource-discovery promitor/promitor-agent-resource-discovery \
               --set azureAuthentication.appId='<azure-ad-app-id>' \
               --set azureAuthentication.appKey='<azure-ad-app-key>'

**Reply by tomkerkhove** [reply]
Can you try using `azureAuthentication.identity.id` & `azureAuthentication.identity.key` instead please?

Normally the old ones should work as well but let's see if that helps.

**Reply by tomkerkhove** [reply]
And what are the above screenshots then? Generated resources from the Helm chart or?

**Comment by LukaszSzalek-TomTom** [comment, marked as answer]
Seems that I had to set
env:
        - name: AUTH_APPKEY
as environment variable not a PROMITOR_AUTH_APPID

**Reply by tomkerkhove** [reply]
Yes, that's a predefined prefix, I should have mentioned that sorry :)

But again, if you use the Helm chart that should work like a charm :) Can you share the output for just relying on the chart please?

**Reply by LukaszSzalek-TomTom** [reply]
AUTH_APPKEY not AUTH_APPID, so small but crucial diff

**Reply by tomkerkhove** [reply]
Again, if you use this for the Helm chart you should be OK: https://github.com/tomkerkhove/promitor/discussions/1879#discussioncomment-1785648
